src/core/scheduler.rs

```diff
@@ -10,10 +10,45 @@ use std::sync::{Arc, Condvar, Mutex};
 use std::thread;
 use std::time::{Duration, Instant};
 
+/// A guard canceling the inner handle when dropped.
+///
+/// See [Cancel::into_guard](trait.Cancel.html#method.into_guard) to create it.
+pub struct CancelGuard<C: Cancel> {
+    inner: Option<C>,
+}
+
+impl<C: Cancel> CancelGuard<C> {
+    /// Disarms the guard.
+    ///
+    /// This disposes of the guard without performing the cancelation. This is similar to calling
+    /// `forget` on it, but doesn't leak resources, while forget potentially could.
+    pub fn disarm(mut self) {
```

Comment:
Maybe return the guarded Handle to let calling code regain explicit control of cancellation.

Also not sure about inner needing to be an `Option` since `disarm` is the only method and consumes the Guard anyway?

Reply:
OK, I can return it back. But I'm not sure if `disarm` is the best name, then. Maybe `into_inner` or `unwrap` or event `into_manual_handle`?

I've added a comment explaining why it needs to be an Option ‒ destructuring is disallowed on types with a destructor :-(.